service/controller/internal/changedetection/tcnp.go

```diff
@@ -93,6 +95,28 @@ func (t *TCNP) ShouldUpdate(ctx context.Context, md infrastructurev1alpha2.AWSMa
 		t.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("detected tenant cluster node pool should update due to operator version changes from %#q to %#q", cc.Status.TenantCluster.OperatorVersion, key.OperatorVersion(&md)))
 		return true, nil
 	}
+	if !securityGroupEqual {
+		t.logger.LogCtx(ctx, "level", "debug", "message", "detected tenant cluster node pool should update due to node pool security groups")
+		return true, nil
+	}
 
 	return false, nil
 }
+
+func securityGroupListEqual(currentSecurityGroupIDs []string, nodePoolsSecurityGroups []*ec2.SecurityGroup) bool {
```

Comment:
here we check if the Node Pools General Security Group from all MachineDeployment/NodePool is included  in the list of ingress security group rules